src/main/java/org/cloudburstmc/server/block/trait/IntegerBlockTrait.java

```diff
@@ -51,14 +51,19 @@ public int getIndex(Object value) {
         return getIndex((int) value);
     }
 
-    public IntRange getRange() {
-        return range;
-    }
-
     public int getIndex(int value) {
         int start = this.range.getStart();
         int end = this.range.getEnd();
         checkArgument(value >= start && value <= end, "value is not in range");
         return value - start;
     }
+
+    public IntRange getRange() {
+        return range;
+    }
+
+    @Override
+    public Integer parseValue(String text) {
```

Comment:
Can we not use primitives? :(

Reply:
I believe the base block trait uses generics, so you can't use primitive type for generics

Reply:
this method is not used by any performance-sensitive code, and it can't be a primitive because other implementations need to return a boolean or enum value